numpy/_core/src/multiarray/descriptor.c

```diff
@@ -3470,7 +3470,7 @@ is_dtype_struct_simple_unaligned_layout(PyArray_Descr *dtype)
         if (key == NULL) {
             return 0;
         }
-        tup = PyDict_GetItem(fields, key);
+        tup = PyDict_GetItem(fields, key); // noqa: borrowed-ref OK
```

Comment:
Do you have a take on this one? I know this is creating a tuple, but I think `fields` uses a borrowed reference `*dtype` - this made me think other threads might have access. 

I was unsure about this one.